components/script/dom/uievent.rs

```diff
@@ -75,15 +75,32 @@ impl UIEvent {
         detail: i32,
         can_gc: CanGc,
     ) -> DomRoot<UIEvent> {
-        let ev = UIEvent::new_uninitialized_with_proto(window, proto, can_gc);
```

Comment:
I think we still want to create an event from `new_uninitialized_with_proto`, and then call `initialize_ui_event`

Reply:
Because the spec requires a plain initialization, which requires `&mut self` for `initialize_ui_event` if we created with `new_uninitialized_with_proto` up front, so I decided to use struct literal to prevent redundant initialization steps for uievent.

Reply:
https://github.com/servo/servo/pull/34524/commits/41e209df4c3cbd7ed067c2fb071562f7903c267e